--- modules/mod_login.php.old	2005-01-05 18:21:11.000000000 +0600
+++ modules/mod_login.php	2005-01-05 18:21:29.000000000 +0600
@@ -75,6 +75,7 @@
 		<input type="submit" name="Submit" class="button" value="<?php echo _BUTTON_LOGIN; ?>" />
 		</td>
 	</tr>
+	<!--
 	<tr>
 		<td>
 		<a href="<?php echo sefRelToAbs( 'index.php?option=com_registration&amp;task=lostPassword' ); ?>">
@@ -82,6 +83,7 @@
 		</a>
 		</td>
 	</tr>
+	-->
 	</table>
 	<?php
 	echo $posttext;
