### Eclipse Workspace Patch 1.0
#P sahana-phase2
Index: inc/lib_stream_image.inc
===================================================================
RCS file: /cvsroot/sahana/sahana-phase2/inc/lib_stream_image.inc,v
retrieving revision 1.4
diff -u -r1.4 lib_stream_image.inc
--- inc/lib_stream_image.inc	18 Feb 2008 11:17:07 -0000	1.4
+++ inc/lib_stream_image.inc	3 Mar 2009 11:11:03 -0000
@@ -42,4 +42,3 @@
 	// do nothing.
 	return;
 }
-?>
Index: inc/handler_login.inc
===================================================================
RCS file: /cvsroot/sahana/sahana-phase2/inc/handler_login.inc,v
retrieving revision 1.14
diff -u -r1.14 handler_login.inc
--- inc/handler_login.inc	30 Jan 2008 04:38:38 -0000	1.14
+++ inc/handler_login.inc	3 Mar 2009 11:11:02 -0000
@@ -19,12 +19,14 @@
 global $conf;
 
 if ($_SESSION['logged_in'] != true ) {
+	//destroy the existing session and start a new.
 ?>
     <div id="loginform">
 	<form method="post" action="index.php?act=login">
         <fieldset><legend><?=_t('Login')?></legend>
         <label><?=_t('User Name')?></label><input type="text" name="user_name" id="userName" />
         <label><?=_t('Password')?></label><input type="password" name="password" id="pswd" />
+        <input type="hidden" name="sec_key" value="<?php echo shn_generate_sec_key();?>"/>
 		<input type="submit" value="Sign In" />
 		<br />
         <!-- <a href="#">Forgot your password?</a> -->
Index: inc/lib_menu.inc
===================================================================
RCS file: /cvsroot/sahana/sahana-phase2/inc/lib_menu.inc,v
retrieving revision 1.20
diff -u -r1.20 lib_menu.inc
--- inc/lib_menu.inc	30 Jan 2008 04:38:38 -0000	1.20
+++ inc/lib_menu.inc	3 Mar 2009 11:11:03 -0000
@@ -28,8 +28,12 @@
 { 
     global $global;
     if ( null == $module ) $module = $global['module'];
+    /*
+     * Check if the menu should be shown to the currenly logged in user.
+     */
+    $allowed = shn_acl_is_menu_item_allowed($action,$module);
 ?>
-    <li><a href="index.php?mod=<?php echo $module;?>&amp;act=<?php echo $action;?>"><?php echo $desc;?></a></li>
+    <li><a href=<?php if($allowed){ ?>"index.php?mod=<?php echo $module;?>&amp;act=<?php echo $action;?>"<?php }else{ echo '"#" class="disabled"';} ?>><?php echo $desc;?></a></li>
 <?php
 }
 
@@ -78,9 +82,13 @@
     global $global;
 
     if ( null == $module ) $module = $global['module'];
+    /*
+     * Check if the menu should be shown to the currenly logged in user.
+     */
+    $allowed = shn_acl_is_menu_item_allowed($action,$module);
     shn_breadcrumb_set_nicename($module, $action, $desc);
 ?>
-    <li><a href="index.php?mod=<?php echo $module;?>&amp;act=<?php echo $action;?>"><?php echo $desc;?></a></li>
+    <li><a href=<?php if($allowed){ ?>"index.php?mod=<?php echo $module;?>&amp;act=<?php echo $action;?>"<?php }else{ echo '"#" class="disabled"'; } ?>><?php echo $desc;?></a></li>
 <?php
 }
 
@@ -128,9 +136,13 @@
 { 
     global $global;
     if ( null == $module ) $module = $global['module'];
+    /*
+     * Check if the menu should be shown to the currenly logged in user.
+     */
+    $allowed = shn_acl_is_menu_item_allowed($action,$module);
     shn_breadcrumb_set_nicename($module, $action, $desc);
 ?>
-		<li><a href="index.php?mod=<?php echo $module;?>&amp;act=<?php echo $action;?>"><?php echo $desc;?></a></li>
+		<li><a href=<?php if($allowed){ ?>"index.php?mod=<?php echo $module;?>&amp;act=<?php echo $action;?>"<?php }else { echo '"#" class="disabled"';} ?>><?php echo $desc;?></a></li>
 <?php
     global $sub_menu_id;
     $sub_menu_id=$action;
@@ -188,9 +200,13 @@
 { 
     global $global;
     if ( null == $module ) $module = $global['module'];
+    /*
+     * Check if the menu should be shown to the currenly logged in user.
+     */
+    $allowed = shn_acl_is_menu_item_allowed($action,$module);
     shn_breadcrumb_set_nicename($module, $action, $desc);
 ?>
-    <a <?=($global['action']==$action?'class="selected" ':'');?> href="index.php?mod=<?=$module?>&amp;act=<?=$action?>"><?=$desc?></a>
+    <a <?php if($allowed){ echo ($global['action']==$action?'class="selected" ':'');?> href="index.php?mod=<?php echo $module?>&amp;act=<?php echo $action?>"<?php }else{ echo 'href="#" class="disabled"';} ?>><?php echo $desc?></a>
 <?php
 }
 
@@ -329,16 +345,25 @@
  * @access public
  * @return void
  */
-function shn_tabmenu_item($action, $desc, $module = null){
+function shn_tabmenu_item($action, $desc, $module = null, $extra_opts = null){
     global $global;
     if ( null == $module ) $module = $global['module'];
+    /*
+     * Check if the menu should be shown to the currenly logged in user.
+     */
+    $allowed = shn_acl_is_menu_item_allowed($action,$module);
+    
     static $count=0;
     $active=($count == $_GET['tabid'])?"id='active'":'';
     $tabid='&tabid='.$count++;
     shn_breadcrumb_set_nicename($module, $action, $desc);
+    $ex_query = '';
+    if(isset($extra_opts['query_string'])){
+    	$ex_query = '&amp;'.$extra_opts['query_string'];
+    }
 ?>
         <li <?php echo $active ?>>
-        <a href="index.php?mod=<?php echo $module?>&amp;act=<?php echo $action.$tabid?>"><span><?php echo $desc?></span></a></li>
+        <a href=<?php if($allowed){ ?>"index.php?mod=<?php echo $module?>&amp;act=<?php echo $action.$tabid.$ex_query?>"<?php }else{ echo '"#" class="disabled"'; } ?>><span><?php echo $desc?></span></a></li>
 <?php
 }
 ?>
Index: inc/handler_mainmenu.inc
===================================================================
RCS file: /cvsroot/sahana/sahana-phase2/inc/handler_mainmenu.inc,v
retrieving revision 1.23
diff -u -r1.23 handler_mainmenu.inc
--- inc/handler_mainmenu.inc	8 Jan 2009 03:23:06 -0000	1.23
+++ inc/handler_mainmenu.inc	3 Mar 2009 11:11:02 -0000
@@ -1,4 +1,4 @@
-<?php
+<?php  
 /**
  * This handler displays the mainmenu dynamically based on action
  * Sahana - http://sahana.sourceforge.net
@@ -33,21 +33,19 @@
 $module_menuorder = array();
 $module_directory = array();
 
-foreach ($module_list as $i) {
-	if(isset($conf['mod_'.$i.'_hide_in_menu'])==false){
-		$tmp = isset($conf['mod_'.$i.'_name'])?$conf['mod_'.$i.'_name']:null;
-		array_push($module_names,( null == $tmp )? $i : $tmp);
-		$tmp = isset($conf['mod_'.$i.'_menuorder'])?$conf['mod_'.$i.'_menuorder']:null;
-		array_push($module_menuorder,( null == $tmp )? 10 : $tmp);
-		$tmp = isset($conf['mod_'.$i.'_directory'])?$conf['mod_'.$i.'_directory']:null;
-		array_push($module_directory, $i );
-	}
-}
+foreach ($module_list as $i) {  
+    $tmp = isset($conf['mod_'.$i.'_name'])?$conf['mod_'.$i.'_name']:null; 
+    array_push($module_names,( null == $tmp )? $i : $tmp);
+    $tmp = isset($conf['mod_'.$i.'_menuorder'])?$conf['mod_'.$i.'_menuorder']:null; 
+    array_push($module_menuorder,( null == $tmp )? 10 : $tmp);
+    $tmp = isset($conf['mod_'.$i.'_directory'])?$conf['mod_'.$i.'_directory']:null; 
+    array_push($module_directory, $i );
+} 
 
 array_multisort($module_menuorder, $module_names, $module_directory);
 
 for ($i = 0; $i < count($module_menuorder); $i++) {
-	shn_mainmenuitem('default',$module_names[$i], $module_directory[$i]);
+    shn_mainmenuitem('default',$module_names[$i], $module_directory[$i]);
 }
 
 shn_mainmenuclose();
Index: inc/lib_form.inc
===================================================================
RCS file: /cvsroot/sahana/sahana-phase2/inc/lib_form.inc,v
retrieving revision 1.87
diff -u -r1.87 lib_form.inc
--- inc/lib_form.inc	26 Jan 2009 06:43:57 -0000	1.87
+++ inc/lib_form.inc	3 Mar 2009 11:11:03 -0000
@@ -905,3 +905,51 @@
     }
             shn_form_extra_opts($extra_opts);
 }
+/**
+ * This function will output an image button for a form.
+ *
+ * @param String $name The name of the button
+ * @param String $image_url The url of the image.
+ * @param String $button_opts Any options for the button such as javascript event handler calls.
+ */
+function shn_form_imagebutton($name,$image_url,$button_opts=null){
+	global $shn_tabindex;
+	?> <input type="image" src="<?php echo $image_url?>"
+	name="<?php echo $name?>" <?php echo $submit_opts?>
+	tabindex="<?php echo ++$shn_tabindex?>" /> <?php
+}
+
+/**
+ * This will write a link secured by the new ACL module.
+ *
+ * @param String $mod The module short code for the link
+ * @param String $act The action of the link
+ * @param String $text The link text.
+ * @param String $stream The stream for the link.
+ * @param Array $params Any additional HTTP Parameters for the link.
+ * @param String $text_opts Text Options
+ * @param Boolean $return_link Whether to return the href link instead of sending it to the output string.
+ */
+function shn_acl_secure_hyperlink($mod='home',$act='default',$text='',$stream=null,$params=null,$text_opts='',$return_link = false){
+	global $shn_tabindex;
+	$param_str = "";
+
+	if(shn_acl_is_hyperlink_allowed($act,$mod,$stream)){
+		foreach($params as $k=>$v){
+			$param_str .= "&$k=$v";
+		}
+
+		$link = "index.php?mod=$mod&act=$act".(($stream!=null)?("&".$stream):"")."$param_str";
+		if($return_link){
+			return $link;
+		}else{
+			echo "<a href=\"$link\" $text_opts tabindex=\"".(++$shn_tabindex)."\">$text</a>";
+		}
+	}else{
+		if($return_link){
+			return '';
+		}else{
+			echo "<a href='javascript:void(0)' class='disabled'>$text</a>";
+		}
+	}
+}
\ No newline at end of file
Index: inc/lib_stream_html.inc
===================================================================
RCS file: /cvsroot/sahana/sahana-phase2/inc/lib_stream_html.inc,v
retrieving revision 1.16
diff -u -r1.16 lib_stream_html.inc
--- inc/lib_stream_html.inc	20 Jul 2008 16:27:47 -0000	1.16
+++ inc/lib_stream_html.inc	3 Mar 2009 11:11:03 -0000
@@ -26,7 +26,8 @@
 	require_once ($global['approot'].'inc/lib_security/lib_acl.inc');
 
 	$module = $global['module'];
-
+	// topmost output buffer.
+	ob_start();
 	// include the html head tags
 	shn_include_page_section('html_head',$module);
 
@@ -135,9 +136,8 @@
 	}
 
 	?></div>
-<!-- /content --> 
-<?php
-    // if xajax object exists print the javascript
+<!-- /content --> <?php
+// if xajax object exists print the javascriheader("Location: index.php?mod=admin&act=acl_settings");pt
     if(isset($global['xajax']))
         shn_xajax_printJavascript();
     // include the footer provided there is not a module override
@@ -148,5 +148,18 @@
 </body>
 </html>
 <?php
+echo "</body>";
+echo "</html>";
+// dump and clean the top most output buffer.
+$all_content=ob_get_contents();
+ob_end_clean();
+//var_dump($global['http_headers']);
+// write the headers if any
+if(isset($global['http_headers'])){
+    foreach($global['http_headers'] as $header){
+        header($header);
+    }    
+}
+echo $all_content;
 }
 ?>
Index: inc/user_feedback.inc
===================================================================
RCS file: /cvsroot/sahana/sahana-phase2/inc/user_feedback.inc,v
retrieving revision 1.8
diff -u -r1.8 user_feedback.inc
--- inc/user_feedback.inc	27 Feb 2008 03:13:56 -0000	1.8
+++ inc/user_feedback.inc	3 Mar 2009 11:11:03 -0000
@@ -104,7 +104,7 @@
 
 	$contents =  ob_get_contents();
 	
-	ob_clean();
+	ob_end_clean();
 	return $contents;
 }
 
@@ -278,7 +278,7 @@
 
 	$contents =  ob_get_contents();
 	
-	ob_clean();
+	ob_end_clean();
 	
 	return $contents;
 }
Index: inc/lib_uuid.inc
===================================================================
RCS file: /cvsroot/sahana/sahana-phase2/inc/lib_uuid.inc,v
retrieving revision 1.32
diff -u -r1.32 lib_uuid.inc
--- inc/lib_uuid.inc	15 Oct 2008 09:41:34 -0000	1.32
+++ inc/lib_uuid.inc	3 Mar 2009 11:11:03 -0000
@@ -189,8 +189,26 @@
 		case 'vm':
 			$gen_id = 'vm-'._shn_gen_id('volunteer');
 			break;
-
+		case 'disease':
+		case 'symptom':
+		case 'patient':
+		case 'diagnosis':
+		case 'dsm':
+			$gen_id = 'dsm-'._shn_gen_id($type);
+			break;
+		case 'sn_category':
+		case 'sn_group':
+		case 'sn_role':
+		case 'sn_forum':
+		case 'sn_topic':
+		case 'sn_post':	
+		case 'sn_group_note':	
+		case 'sn_forum_note':
+		case 'sn_user_note':			
+			$gen_id ='sn-'. _shn_gen_id($type);
+			break;
 		default :
+			$gen_id = 'undefined-'._shn_gen_id($type);
 			break;
 	}
 
Index: inc/lib_errors.inc
===================================================================
RCS file: /cvsroot/sahana/sahana-phase2/inc/lib_errors.inc,v
retrieving revision 1.29
diff -u -r1.29 lib_errors.inc
--- inc/lib_errors.inc	7 Jul 2008 13:13:20 -0000	1.29
+++ inc/lib_errors.inc	3 Mar 2009 11:11:02 -0000
@@ -200,7 +200,7 @@
     
 }
 
-function shn_error_get_restricted_access_message(){
+function shn_error_get_restricted_access_message($msg=null){
 	ob_start();
 	?>
 <p><em><?php echo _t('Sorry, you do not have permisssion to access this section')?></em><br />
@@ -216,6 +216,7 @@
 <?php
 $contents = ob_get_contents();
 ob_end_clean();
+return $contents;
 }
 
 function shn_error_dummy_header()
Index: conf/sysconf.inc.php
===================================================================
RCS file: conf/sysconf.inc.php
diff -N conf/sysconf.inc.php
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ conf/sysconf.inc.php	1 Jan 1970 00:00:00 -0000
@@ -0,0 +1,177 @@
+<?php
+/**
+* The main Sahana configuration file
+*
+* PHP version 4 and 5
+*
+* LICENSE: This source file is subject to LGPL license
+* that is available through the world-wide-web at the following URI:
+* http://www.gnu.org/copyleft/lesser.html
+*
+* @package    Sahana - http://sahana.sourceforge.net
+* @author     
+* @copyright  Lanka Software Foundation - http://www.opensource.lk
+*/
+
+######################################################################
+#                 Sahana Main Configuration Settings                 #
+######################################################################
+#
+#
+# Welcome to Sahana the disaster management system.
+# feel free to edit this config file.
+# if you have problem please visit http://sahana.sourceforge.org/
+# or join with our irc channel #sahana at freenode.org, 
+
+# Specify the name of this Sahana instance. This should be a unique identifier
+# of this instance of Sahana. 
+# It has to be a 4 character alphanumeric 
+$conf['base_uuid'] = 't8c0';
+
+# Disable the access control system
+$conf['acl_base'] = false;
+
+# Root Name :The owner of the machine
+$conf['root_name'] = 'sandaruwan';
+
+# Root Email :The email address of the admin
+$conf['root_email'] = '';
+
+# Root Telephone :The telephone of the admin
+$conf['root_tel'] = '';
+
+
+
+
+# specify the host ip address of the database reside.
+# if it's the same server that Sahana reside then put 'localhost'
+#
+$conf['db_host'] = 'localhost';
+
+# port that data base talks. leave blank for default.
+#
+$conf['db_port'] = '3306';
+
+# theme that sahana will use todo
+#
+$conf['theme'] = 'default';
+
+# specify the database name.
+#
+$conf['db_name'] = 'sahana_phase2';
+
+# specify user name that Sahana can use to connect.
+#
+$conf['db_user'] = 'sahana_phase2';
+
+# And password for that user.
+#
+$conf['db_pass'] = 'password';
+
+#debug variable
+# true/false
+$conf['debug'] = false;
+
+##########################
+# Database Configuration #
+########################## 
+
+# Session writer 
+# enter your database name here.
+#
+$conf['session_writer'] = 'database' ; 
+
+# Sahana uses data base abstraction layer for connecting to data base.
+# specify the Database Abstraction Layer Library Name here.
+# Database Abstraction Layer Libraries are reside in 
+# /inc/lib_database/db_libs/
+# The name should be same as the library folder
+#
+$conf['dbal_lib_name'] = 'adodb' ;
+
+# mention the database engine name
+# @todo Find supported engine list
+# for the moment, Sahana supported and tested on PostgreSQL and MySql
+#
+# $conf['db_engine'] = 'postgres'; 
+$conf['db_engine'] = 'mysql';
+
+#specify the mysql engine to be used
+$conf['storage_engine'] = '';
+
+# enable monitor time that takes to process sql queries
+# this is an advance feature and recommended only for developers
+#
+$conf['enable_monitor_sql'] = 'true';
+
+# @todo Look into the database caching directories etc
+# This is a testing feature.
+#
+$conf['enable_cache'] = false;
+$conf['cache_dir'] = 'cache/db_cache';
+
+# Default locale
+#
+$conf['locale'] = 'en_US';
+
+# Logging Configuration
+#
+#$conf['default_logger'] = 'DatabaseLogger';
+$conf['default_logger'] = 'FileLogger';
+#File Logger Specific Configuration.
+#Prefix to the log file name
+$conf['log_file_name_prefix'] = 'log';
+#Log file location relative to approot.
+$conf['log_file_location'] = "logs";
+
+
+###############################
+# GIS and Mapping Configuration
+###############################
+# See gis_conf.inc for more configuration information
+
+# GIS Functionality
+# true: to enable GIS/Mapping Capabilities
+$conf['gis'] = true;
+
+$conf['proxy_path']='res/lib_proxy.php?url=';
+
+###############################
+
+###############################
+#   Help and Wiki Urls        #
+###############################
+# Default values are given below.
+$conf['wiki_url'] = 'http://wiki.sahana.lk/doku.php?id=doc:nwhome';
+
+$conf['sahana_url'] = 'http://sahana.lk';
+
+$conf['forum_url'] = 'http://forum.sahana.lk/';
+
+$conf['chat_url'] = 'http://www.sahana.lk/chat';
+
+$conf['shn_user_feedback_enabled'] = false;
+
+
+$conf['user_help_server'] = 'localhost';
+
+$conf['use_local_help'] = false;
+#####################################
+#   Browser Capabilities Settings   #
+#####################################
+// whether to auto update the browscaps.ini file
+$conf['bcaps_auto_update'] = false;
+
+
+
+#####################################
+#      Miscellaneous Options        #
+#####################################
+
+// which wysiwyg you want to use leave blank for default whcih is EXTjs editor
+// available options ( 'fckeditor' , 'tiny_mce' )
+$conf['wysiwyg']= '';
+
+
+# end of the config file.
+
Index: .cvsignore
===================================================================
RCS file: .cvsignore
diff -N .cvsignore
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ .cvsignore	1 Jan 1970 00:00:00 -0000
@@ -0,0 +1,1 @@
+.project
